app/controllers/api/v1/schedules_controller.rb

```diff
@@ -37,12 +37,7 @@ def availableTimeSlots
     api :POST, "/v1/schedules", "Make a booking for the pick up"
     param_group :schedule
     def create
-      @schedule.attributes = schedule_params
-      if @schedule.save
-        render json: @schedule, serializer: serializer, status: 201
-      else
-        render json: @schedule.errors.to_json, status: 422
-      end
+      assign_params_and_render_object(@schedule, serializer, schedule_params)
```

Comment:
the `assign_params...` method uses the `cached_json` model but that isn't here in schedule controller and might produce unexpected effects.

The `data_object = data_object.find( params[:ids].split(",") ) if params[:ids].present?` line in `assign_params...` function is not something that happens in the `schedules#create` controller - possible unintended consequences?